qiskit/quantum_info/operators/symplectic/clifford.py

```diff
@@ -186,15 +186,6 @@ def paulis(self):
         """Return PauliList"""
         return self._paulis
 
-    @paulis.setter
-    def paulis(self, value):
-        """Set the Pauli list"""
-        if not isinstance(value, PauliList):
-            value = PauliList(value)
-        self._paulis._z[:, :] = value._paulis._z
-        self._paulis._x[:, :] = value._paulis._x
-        self._paulis._phase[:] = value._paulis._phase
-
```

Comment:
If you're getting rid of the setter entirely, I think you'd need to get rid of the regular property as well, and make `self._paulis` just be called `self.paulis` in the `__init__`.  You don't need to make such a change, though - I think here, all I was suggesting was just that you replaced the last three lines
```python
        self._paulis._z[:, :] = value._paulis._z
        self._paulis._x[:, :] = value._paulis._x
        self._paulis._phase[:] = value._paulis._phase
```
with simply
```python
       self._paulis = value
```

I think you still need a way of setting the Paulis, because your other deprecation warnings are telling people to use that setter method.

Reply:
Perhaps the intent of the original code was to copy. so I'll fix it to
```
self._paulis = value.copy()
```

Reply:
Added by fb934b7

Reply:
The intent of the original code was to prevent changing the size of the arrays. If you set the pauli variable directly you can invalidate the clifford by changing qubit number or having wrong number of elements in the list, so you would need to add additional validation.

Honestly we probably shouldn't have a setter for this at all and you should just construct a new clifford as `Clifford(pauli_list)` if you are changing the entire structure. The init will at least validate that the pauli list corresponds to a valid clifford.

@jakelishman I think the point of keeping `paulis` as a property without a setter rather than an attribute is to explicitly prevent setting it to a new value.

Reply:
I see. we can certainly guarantee the size by this way. Then I feel like we can just revert this commit fb934b7.

Reply:
Sorry, I'm not actually at work today, and last night I got distracted by a critical bug that needed fixing for qiskit-experiments.

I'm happy with whatever setter methods you guys want, as long as it's clear in the deprecation warning what the alternative is.  If there is to be no setter for `paulis` on the basis that you can do `obj.paulis[:] = ...`, then I think we need tests to make sure that this continues to work; because `paulis` is a `property` itself, it's very easy to accidentally make changes such that assigning to it.

Reply:
Thanks. I know you're busy with other PRs.
That should be ensured by testing on the PauliList side, not Clifford. Such a test already exist　https://github.com/Qiskit/qiskit-terra/blob/main/test/python/quantum_info/operators/symplectic/test_pauli_list.py#L432.